bip-PC.md

```diff
@@ -0,0 +1,187 @@
+<pre>
+  BIP: ?
+  Layer: Consensus (soft fork)
+  Title: OP_PAIRCOMMIT
+  Author: moonsettler <moonsettler@protonmail.com>
+  Comments-Summary: No comments yet.
+  Comments-URI: <links to wiki page for comments>
+  Status: Draft
+  Type: Standards Track
+  Created: 2024-11-08
+  License: BSD-3-CLAUSE
+</pre>
+
+## Abstract
+
+This BIP describes a new tapscript opcode `OP_PAIRCOMMIT` which
+provides limited vector commitment functionality in tapscript.
+
+When evaluated, the `OP_PAIRCOMMIT` instruction:
+* Pops the top two values off the stack,
+* takes the "PairCommit" tagged SHA256 hash of the stack elements,
+* pushes the resulting commitment on the top of the stack.
+
+## Motivation
+
+To do LN-Symmetry contracts that don't require the nodes to keep old states,
+we need to solve the data availability problem presented by unilateral closes.
+Channel peers must be able to reconstruct the script that spends an
+intermediate state.
```

Comment:
what data needs to be available? how does PC solve that problem (does it stick the data in the witness and put a commitment somewhere covered by a signature? something else?)? Is this mechanism useful for things outside of LN-Symmetry?

Reply:
The data that needs to be available for state n is:
```text
state-n-recovery-data { settlement-n-hash or state-n-balance }
```
This is needed to reconstruct the whole script for the nth state address that the funds move to by the channel peer that only holds the latest state, so he can spend to the latest state.

Reply:
Updated. I believe we can resolve these conversations? @rot13maxi @murchandamus